src/WeakValueDict.jl

```diff
@@ -145,26 +511,34 @@ function Base.get(wkh::WeakValueDict{K, V}, key, default) where {K, V}
         if x === default
             return x
         else
-            return x.value::V
+            y = x.value
+            if y === nothing
+                return default
+            else
+               return y::V
+            end
         end
     end
 end
+
+#### the rest of these look dodgy
+
 function Base.get(default::Base.Callable, wkh::WeakValueDict{K}, key) where {K}
     key === nothing && throw(KeyError(nothing))
     lock(wkh) do
-        return get(default, wkh.ht, key)
+        return get(default, wkh.ht, key)  # ???
     end
 end
 function Base.pop!(wkh::WeakValueDict{K}, key) where {K}
     key === nothing && throw(KeyError(nothing))
     lock(wkh) do
-        return pop!(wkh.ht, key).value
+        return pop!(wkh.ht, key).value  # ???
```

Comment:
?????

Reply:
Is it ok for `pop!` to return nothing? Do we keep `pop!`'ing until we the poped value is not nothing or we can't `pop!` anymore? These are the kind of questions imposed by fake dictionaries...

Reply:
If the value is nothing at this point, then this is equivalent to "the mapping does not exist", in which case the documentation says
> throw an error if default is not specified